EditorControls/DropDownTypesControl.xaml.cs

```diff
@@ -116,6 +116,14 @@ public void Save()
 
             m_helper.Controller.StartTransaction(String.Format("Change type from '{0}' to '{1}'", m_dropDownValues[m_currentType], m_dropDownValues[selectedType]));
 
+            if (selectedType == k_noType)
+            {
+              if (m_currentType == "editor_object" ||  m_currentType == "editor_room")
+                {
+                  m_helper.Controller.AddFieldToElement(m_data.Name, "isroom", true);
+                }
+            }
+            

```

Comment:
This is feeling clunky to me. This is changing the dropdown types control to have some special cases to set other fields on an element. It's a sign that something is wrong with this design because logically the drop down types control shouldn't need to know anything about what a room is, or changing other fields on an element.